lib/main.ts

```diff
@@ -565,6 +566,69 @@ export async function unviewConfession(
   }
 }
 
+export async function deleteInactiveRecords(
+  repository: Repository<Confession>,
+  record: Confession,
+): Promise<boolean> {
+  const now = new Date();
+  const weekAgo = now.setDate(now.getDate() - 7) / 1000;
+  const old_published_ts = record.published_ts;
+  if (!old_published_ts) {
+    console.log(`Deleting salt and hashes for the records that are staged and dead...`);
+    try {
+      record.uid_salt = "empty"
+      record.uid_hash = "empty hash";
+      await repository.save(record);
+      return true;
+    } catch (_) {
+      throw `Failed to update Postgres record`;
+    }
+  } else if (old_published_ts) {
+    console.log(`Deleting the salt and hashes which are dead, but were published...`);
+    const thread_messages = await web.conversations.replies({
+      channel: confessions_channel,
+      ts: record.staging_ts as string,
+      limit: 1,
+      latest: "now"
+    });
+
+    if (thread_messages.messages) {
+      const message = thread_messages.messages[0];
+      try {
+        if (!message.latest_reply) {
+          try {
+            record.uid_salt = "empty"
+            record.uid_hash = "empty hash";
```

Comment:
Alternatively, using all zeros would be okay too
```suggestion
            record.uid_salt = ""
            // hashUser("", "")
            record.uid_hash = "d72c87d0f077c7766f2985dfab30e8955c373a13a1e93d315203939f542ff86e73ee37c31f4c4b571f4719fa8e3589f12db8dcb57ea9f56764bb7d58f64cf705";
```